Support for lambda GEARMAN_JOB_NAME.

toudi opened the pull request:
Hello.

This patch solves the issue with the task name being not unique. I must state however, that i am not satisfied with how i explained it in the readme file. I know what i want to describe, but somehow fail to do that in a really easy way :( I would be very grateful for some advice on this matter.

Commit:
Support for lambda GEARMAN_JOB_NAME.

fwenzel commented on django_gearman/models.py:
```diff
@@ -2,6 +2,14 @@
+
+
+def default_taskname_decorator(task_name):
+    return str(adler32(getcwd()) & 0xffffffff) + "." + task_name
```
Why adler32?

Also, you should not use `+` for string concatenation. Use string interpolation or `''.join(...)`.

toudi replied:
As python's doc state:

>  (An Adler-32 checksum is almost as reliable as a CRC32 but can be computed much more quickly.)

as for string concatenation.. ''.join() for two strings ? Apart from the fact, that it doesn't produce readable code, it seems like an overkill ;). Also, my naive thinking tells me that using string interpolation ( formatting ?) requires more CPU cycles than simple addition. 

Why isn't `+` good? (i'm not trying to argue here, just trying to know python better).

fwenzel replied:
Okay sounds good (re: Adler32).

Yeah for two strings it's a wash, but it's three strings here. + is not usually used in Python for concatenating 3 or more strings because every time + is used, the interpreter makes a new string and copies both sides into it. If you do this 1000 times, you'll create 999 throw-away strings. As you say, with very few strings it's not a big deal, but it's "good Python style" to use string interpolation or join() on three or more strings.

toudi replied:
i just tried http://www.python.org/dev/peps/pep-0215/ (string interpolation) and i get syntax error. The page states that it is Superseded by pep 0292. Would you like me to use pep 0292 or string formatting ( like 'foo%s bar' ) ?

fwenzel replied:
String formatting should be fine! `'%s.%s' % (str(adler...), task_name)` should work.

fwenzel commented on django_gearman/models.py:
```diff
@@ -54,6 +62,9 @@ def submit_job(
+        
+        if callable(task_name_decorator):
+            task = task_name_decorator(task)
         """
```
Oops, don't insert code _before_ the docstring. Put the docstring back to the top and add the code afterwards.

fwenzel commented on django_gearman/models.py:
```diff
@@ -91,3 +102,10 @@ def __init__(self, **kwargs):
                 settings.GEARMAN_SERVERS, **kwargs)
+    
+    def register_task(self, task_name, task):
+        
```
I'd remove the empty line after the function definition.

fwenzel commented on django_gearman/models.py:
```diff
@@ -2,6 +2,14 @@
 import gearman
 from django.conf import settings
+from zlib import adler32
+from os import getcwd
```
Fix the import order: "import"s before "from" imports, and stdlib on top, third-party imports afterwards. So in this case:

```
from os...
from zlib...

import gearman
from django....
```

fwenzel commented:
Very nice, thank you! I can help fix the readme a bit. I agree that it's hard to explain shortly.

toudi committed:
Changes after code review:
- string formatting instead of '+' operator
- fixed import order
- removed empty lines

toudi committed:
Removed debug

fwenzel commented on django_gearman/models.py:
```diff
@@ -53,11 +60,14 @@ def parse_data(self, arg, args=None, kwargs=None, *arguments, **karguments):
         self, task, orig_data = None, unique=None, priority=None,
         background=False, wait_until_complete=True, max_retries=0,
-        poll_timeout=None, args=None, kwargs=None, *arguments, **karguments):
+        poll_timeout=None, args=None, kwargs=None, *arguments, **karguments):        
```
No trailing whitespace, please.

But I can merge it, then fix that.

fwenzel commented:
Thank you! There are a few tiny style changes I'd like to see but I'll add those after I merge. Sorry for the delay!!
